### PR TITLE
[CSR] Clarify implicit var convention

### PR DESCRIPTION
**Pull request by flomnes**

### CODE REVIEW

File: src/solver/optimisation/hourly_csr_problem.h [hunk 1]
```diff
@@ -40,5 +40,17 @@ class HOURLY_CSR_PROBLEM
...
+        {
+        }
+        LinkVariable(int direct, int indirect) : directVar(direct), indirectVar(indirect)
+        {
+        }
+        int directVar;
+        int indirectVar;
+    };
+    std::map<int, LinkVariable>
+      linkSet; // place inside only links between to zones inside adq-patch
```
Comment (guilpier-code) [inline src/solver/optimisation/hourly_csr_problem.h]:
```C++
//... links between 2 (or two) zones inside ...
```

**Review by payetvin — APPROVED** [review]

File: src/solver/optimisation/adequacy_patch_csr/solve_problem.cpp [hunk 2]
```diff
@@ -180,27 +180,39 @@ double calculateCsrCostFunctionValue(const PROBLEME_ANTARES_A_RESOUDRE* Probleme
...
-            logs.debug() << "CoutQ: " << ProblemeAResoudre->CoutQuadratique[Var]*1e3;
-            logs.debug() << "TotalCost: " << cost*1e3;
+            logs.debug() << "X-Q: " << ProblemeAResoudre->X[Var] * 1e3;
+            logs.debug() << "CoutQ: " << ProblemeAResoudre->CoutQuadratique[Var] * 1e3;
+            logs.debug() << "TotalCost: " << cost * 1e3;
         }
-        bool inLinkSet = hourlyCsrProblem.linkSet.find(Var) != hourlyCsrProblem.linkSet.end();
-        if (inLinkSet
-            && hourlyCsrProblem.problemeHebdo->adqPatchParams->IncludeHurdleCostCsr)
+        auto itLink = hourlyCsrProblem.linkSet.find(Var);
```
Comment (guilpier-code) [inline src/solver/optimisation/adequacy_patch_csr/solve_problem.cpp]: **linkSet** is vague. Prefer **linksInsideAdqPatch** for instance
Same remark for **ensSet**

File: src/solver/optimisation/hourly_csr_problem.h [hunk 3]
```diff
@@ -40,5 +40,17 @@ class HOURLY_CSR_PROBLEM
...
-    std::set<int> linkSet; // place inside only links between to zones inside adq-patch
+    struct LinkVariable
+    {
+        LinkVariable() : directVar(-1), indirectVar(-1)
+        {
+        }
+        LinkVariable(int direct, int indirect) : directVar(direct), indirectVar(indirect)
+        {
+        }
+        int directVar;
```
Comment (guilpier-code) [inline src/solver/optimisation/hourly_csr_problem.h:78]: I suppose we need the 2 data members  **directVar** or **indirectVar**, and that only 
```C++
bool directVar;
``` 
would not have been sufficient

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/solver/optimisation/hourly_csr_problem.h (hunk 3 above)
Comment (flomnes) [reply src/solver/optimisation/hourly_csr_problem.h:78]: No, for each alg. flux variable, we need the corresponding direct & indirect flux variables

File: src/solver/optimisation/hourly_csr_problem.h (hunk 1 above)
Comment (flomnes) [reply src/solver/optimisation/hourly_csr_problem.h]: done

File: src/solver/optimisation/adequacy_patch_csr/solve_problem.cpp (hunk 2 above)
Comment (flomnes) [reply src/solver/optimisation/adequacy_patch_csr/solve_problem.cpp]: done

**Review by payetvin — APPROVED** [review]

**Review by guilpier-code — APPROVED** [review]

**Pull request merged**